File: apps/admin_mobile/lib/infrastructure/api/mappers/service_mapper.dart
import '../../../domain/entities/service.dart';

Service mapService(Map<String, dynamic> json) {
  return Service(
    id: json['id'] as int,
    name: json['name'] as String,
    description: json['description'] as String?,
    price: (json['price'] as num).toDouble(),
    isActive: json['is_active'] as bool? ?? true,
    imageUrl: json['image_url'] as String?,
    sortOrder: json['sort_order'] as int? ?? 0,
    createdAt: DateTime.parse(json['created_at'] as String),
  );
}

File: apps/admin-v2/src/infrastructure/api/mappers/service.mapper.ts
import type { Service } from '@/domain/entities/service';

export function mapService(raw: Record<string, unknown>): Service {
  return {
    id: raw.id as string,
    name: raw.name as string,
    description: (raw.description as string) ?? null,
    price: typeof raw.price === 'string' ? parseFloat(raw.price) : (raw.price as number),
    isActive: (raw.is_active as boolean) ?? true,
    imageUrl: (raw.image_url as string) ?? null,
    sortOrder: (raw.sort_order as number) ?? 0,
    createdAt: new Date(raw.created_at as string),
  };
}
